### PR TITLE
Implement reporter OnTestRunActionDone/Start methods (closes #4507)

### LINKED ISSUE #4507: Implement onActionDone

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

File: src/role/index.js [hunk 1]
```diff
@@ -34,15 +34,16 @@ class Role extends EventEmitter {
 
     async _executeInitFn (testRun) {
         try {
-            testRun.disableDebugBreakpoints = false;
-            await this.initFn(testRun);
+            let fn = () => this.initFn(testRun);
+
+            fn = testRun.decoratePreventEmitActionEvents(fn, false);
```
Comment (miherlosev) [inline src/role/index.js]: It's difficult to read this construction.
Change as follows:
```js
fn = testRun.decoratePreventEmitActionEvents(fn, { prevent: false });
fn = testRun.decorateDisableDebugBreakpoints(fn, { disable: false });
```

File: test/functional/fixtures/reporter/reporter.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    async reportTestDone () {
+    },
+    async reportTaskDone () {
+    }
+};
+
+function generateReport (log, emitOnStart, emitOnDone, includeBrowserInfo, includeTestInfo) {
+    return function () {
+        return Object.assign({}, baseReport, {
+            async reportTestRunActionStart (name, { browser, test }) {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/reporter/reporter.js]: `reportTestActionStart`

File: test/functional/fixtures/reporter/reporter.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                if (includeBrowserInfo)
+                    item.browser = browser.alias.split(':')[0];
+
+                if (includeTestInfo)
+                    item.test = test;
+
+                log.push(item);
+            },
+
+            async reportTestRunActionDone (name, { command, test, errors }) {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/reporter/reporter.js]: `reportTestActionDone`

File: src/reporter/index.js [hunk 4]
```diff
@@ -177,6 +198,22 @@ export default class Reporter {
...
+
+                await this.plugin.reportTestActionStart(apiActionName, args);
+            }
+        });
+
+        task.on('test-action-done', async ({ apiActionName, command, testRun, errors }) => {
+            if (this.plugin.reportTestActionDone) {
+                const args = this._prepareReportTestActionEventArgs({ command, testRun, errors });
+
+                await this.plugin.reportTestActionDone(apiActionName, args);
```
Comment (AndreyBelym) [inline src/reporter/index.js]: Shouldn't we provide a default implementation for `reportTestActionStart` and `reportTestActionDone` in [src/reporter/plugin-host.js](https://github.com/DevExpress/testcafe/blob/master/src/reporter/plugin-host.js) or check for their existence?

**Pull request merged**